### PR TITLE
[fix][admin]allowAutoTopicCreation and allowAutoTopicCreationType etc should be dynamically configured

### LINKED ISSUE #16020: allowAutoTopicCreation and allowAutoTopicCreationType  etc  should be dynamically configured
**Is your enhancement request related to a problem? Please describe.**
In the broker.conf configuration, there are these configurations:

- allowAutoTopicCreation 
- allowAutoTopicCreationType 
- defaultNumPartitions
- maxNumPartitionsPerPartitionedTopic
- brokerDeleteInactiveTopicsEnabled  
- brokerDeleteInactiveTopicsFrequencySeconds 
- brokerDeleteInactiveTopicsMaxInactiveDurationSeconds
- brokerDeleteInactivePartitionedTopicMetadataEnabled
- brokerDeleteInactiveTopicsMode
- allowAutoSubscriptionCreation

I think these should be dynamically configurable.

**Describe the solution you'd like**
can use ` pulsar-admin brokers update-dynamic-config --config xxx --value xxx`

**Describe alternatives you've considered**
Pulsar has  namespace-level policy, but the broker-level configuration is missing.

**Additional context**
I'm more than willing to fix this, please assign me if it needs to be done

### PR DESCRIPTION
**Pull request by Radiancebobo**
allowAutoTopicCreation and allowAutoTopicCreationType etc should be dynamically configured

Fixes #16020 

### Motivation
In the broker.conf configuration, there are these configurations:

- allowAutoTopicCreation
- allowAutoTopicCreationType
- defaultNumPartitions
- maxNumPartitionsPerPartitionedTopic
- brokerDeleteInactiveTopicsEnabled
- brokerDeleteInactiveTopicsFrequencySeconds
- brokerDeleteInactiveTopicsMaxInactiveDurationSeconds
- brokerDeleteInactivePartitionedTopicMetadataEnabled
- brokerDeleteInactiveTopicsMode
- allowAutoSubscriptionCreation

I think these should be dynamically configurable.

Pulsar has namespace-level policy, but the broker-level dynamically  configuration is missing. 

### Modifications
The change is very simple, just add dynamic identification to these configurations in ServiceConfiguration.java, and add a callback function when updating.

### Documentation

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
The change looks good. Could you please add a UT for the new change? 

Here is an example PR https://github.com/apache/pulsar/pull/13121 which added `subscriptionTypesEnabled` dynamic configuration support.

**Comment by Radiancebobo** [comment]
> The change looks good. Could you please add a UT for the new change?
> 
> Here is an example PR #13121 which added `subscriptionTypesEnabled` dynamic configuration support.

OK, I will supplement the corresponding unit tests as soon as possible

**Comment by Radiancebobo** [comment]
Unit tests that fail to run also happen on the master branch, it doesn't look like it's caused by my commits, I'll try my best to take the time to fix them.

**Review by AnonHxy — APPROVED** [review]

**Review by Demogorgon314 — COMMENTED** [review]
Overall LGTM, just left some small comments.

**Review by Jason918 — APPROVED** [review]
LGTM

**Comment by Radiancebobo** [comment]
org.apache.pulsar.broker.admin.AdminApiTest ► testGetPartitionedStatsInternal：
image
Successfully run multiple times in my local environment：
image


org.apache.pulsar.broker.service.PersistentDispatcherFailoverConsumerTest ► testAddRemoveConsumer：
image
Successfully run multiple times in my local environment：
image


org.apache.pulsar.metadata.ZKSessionTest ► testSessionLost：
image
Runs successfully many times in my local environment, but occasionally fails:
image
image

**Comment by Demogorgon314** [comment]
@codelipenghui @BewareMyPower Please help review this PR. The required test has already passed.

**Comment by BewareMyPower** [comment]
Please rebase to master to resolve conflicts.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java [hunk 1]
```diff
@@ -596,4 +597,64 @@ public void testHealthTopicInactiveNotClean() throws Exception {
...
+        conf.setBrokerDeleteInactivePartitionedTopicMetadataEnabled(false);
+        super.baseSetup();
+        admin.brokers()
+                .updateDynamicConfiguration("brokerDeleteInactivePartitionedTopicMetadataEnabled",
+                        "true");
+        Thread.sleep(2000);
+        Awaitility.await().untilAsserted(()->{
+            assertTrue(conf.isBrokerDeleteInactivePartitionedTopicMetadataEnabled());
+        });
+    }
```
Comment (BewareMyPower) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:657]: Don't use `Thread.sleep`, use `Awaitility.await().atMost(/* ... */)`.

I noticed some existing tests might also use `Thread.sleep`, but it's not good. For new tests, it's better not to sleep.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerServiceAutoSubscriptionCreationTest.java [hunk 2]
```diff
@@ -152,4 +154,24 @@ public void testNonPersistentTopicSubscriptionCreationWithAutoCreationDisable()
...
+        admin.brokers().updateDynamicConfiguration("allowAutoSubscriptionCreation", "false");
+        String topicString = "persistent://prop/ns-abc/non-partitioned-topic" + UUID.randomUUID();
+        String subscriptionName = "non-partitioned-topic-sub";
+        admin.topics().createNonPartitionedTopic(topicString);
+        try {
+            pulsarClient.newConsumer().topic(topicString).subscriptionName(subscriptionName).subscribe();
+            fail("Subscribe operation should have failed");
+        } catch (Exception e) {
+            assertTrue(e instanceof PulsarClientException);
+        }
```
Comment (BewareMyPower) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BrokerServiceAutoSubscriptionCreationTest.java:171]: We can simplify this type of verification to

```java
        Assert.assertThrows(PulsarClientException.class, () ->
                pulsarClient.newConsumer().topic(topicString).subscriptionName(subscriptionName).subscribe());
```

I think we can also apply similar changes to `testDynamicConfigurationTopicAutoCreationDisable`.

File: conf/client.conf [hunk 3]
```diff
@@ -22,12 +22,12 @@
...
 # For TLS:
 # webServiceUrl=https://localhost:8443/
-webServiceUrl=http://localhost:8080/
+webServiceUrl=http://localhost:8081/
 
 # URL for Pulsar Binary Protocol (for produce and consume operations)
 # For TLS:
 # brokerServiceUrl=pulsar+ssl://localhost:6651/
-brokerServiceUrl=pulsar://localhost:6650/
+brokerServiceUrl=pulsar://localhost:6651,localhost:6652,localhost:6653/
```
Comment (BewareMyPower) [inline conf/client.conf:30]: Please revert unnecessary changes.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java [hunk 4]
```diff
@@ -650,7 +650,6 @@ public void testBrokerDeleteInactivePartitionedTopicMetadataEnabled() throws Exc
         admin.brokers()
                 .updateDynamicConfiguration("brokerDeleteInactivePartitionedTopicMetadataEnabled",
                         "true");
-        Thread.sleep(2000);
         Awaitility.await().untilAsserted(()->{
             assertTrue(conf.isBrokerDeleteInactivePartitionedTopicMetadataEnabled());
         });
```
Comment (BewareMyPower) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/InactiveTopicDeleteTest.java:655]: I mean, use `atMost` method to control the timeout explicitly. See https://www.javadoc.io/doc/org.awaitility/awaitility/3.0.0/org/awaitility/Awaitility.html

File: conf/client.conf (hunk 3 above)
Comment (Radiancebobo) [reply conf/client.conf:30]: oh, sorry.

**Review by BewareMyPower — COMMENTED** [review]
Could you add `atMost` to other tests of `InactiveTopicDeleteTest` as well?

**Comment by Radiancebobo** [comment]
> Could you add `atMost` to other tests of `InactiveTopicDeleteTest` as well?

ok

**Review by BewareMyPower — APPROVED** [review]

**Comment by Radiancebobo** [comment]
@BewareMyPower  Can you help me merge this branch to master?

**Pull request merged**